dist/tools/testrunner/testrunner.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+#               2014 Martine Lenders <mlenders@inf.fu-berlin.de>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os, signal, sys, subprocess
+from pexpect import spawnu, TIMEOUT, EOF
+
+def run(testfunc, timeout=5):
+    env = os.environ.copy()
+    child = spawnu("make term", env=env, timeout=timeout)
+    child.logfile = sys.stdout
+
+    try:
+        subprocess.check_output(('make', 'reset'), env=env,
+                                stderr=subprocess.PIPE)
+    except subprocess.CalledProcessError:
+        # make reset yields error on some boards even if successful
+        pass
+    try:
+        testfunc(child)
+    except TIMEOUT:
+        print("Timeout in expect script")
+        return 1
+    finally:
+        print("")
+        child.kill(signal.SIGKILL)
```

Comment:
This won't cut it for the bug mentioned below. Use `pgkill` with the childs PID.